### PR TITLE
LDrawLoader: Make lines honor material opacity

### PR DESCRIPTION
**Pull request by yomboprime**
This is noticeable in the car headlights, or the X-Wing exhausts for example.

Current version: https://raw.githack.com/mrdoob/three.js/dev/examples/webgl_loader_ldraw.html

### CODE REVIEW

**Comment by yomboprime** [comment]
It seems it causes problems with transparency ordering. Glass surfaces hide the lines behind them (noticeable in the car windscreen)
What do you think @gkjohnson ?

**Comment by gkjohnson** [comment]
Awesome! It looks like the lines disappear because `depthWrite` is set to `true` on the transparent materials so they fail the depth test don't get drawn. If `depthWrite` is set to `false` on the transparent materials then the lines show more like expected.

**Comment by WestLangley** [comment]
For transparent materials, you can try setting `depthWrite` to `false`. You can also control `mesh.renderOrder`.

For transparent meshes such as your windshield, it is a bit more complicated: you don't have as much control over render order since you are rendering multi-material meshes.

**Comment by WestLangley** [comment]
Perfect timing. :-)

File: examples/js/loaders/LDrawLoader.js [hunk 1]
```diff
@@ -1288,7 +1288,11 @@ THREE.LDrawLoader = ( function () {
 			if ( ! edgeMaterial ) {
 
 				// This is the material used for edges
-				edgeMaterial = new THREE.LineBasicMaterial( { color: edgeColour } );
+				edgeMaterial = new THREE.LineBasicMaterial( {
+					color: edgeColour,
+					transparent: isTransparent,
```
Comment (gkjohnson) [inline examples/js/loaders/LDrawLoader.js]: Don't forget to set transparency on the conditional edge material, as well!

**Comment by yomboprime** [comment]
@WestLangley @gkjohnson Thanks! Is it right like this?
There are some transparency artifacts in windscreens but I think it is okay.

**Comment by WestLangley** [comment]
I would be careful about turning off depthWrite for transparent meshes as they will no longer depth-test against themselves.

**Comment by mrdoob** [comment]
Looking good!

**Comment by mrdoob** [comment]
Thanks!

**Comment by yomboprime** [comment]
> I would be careful about turning off depthWrite for transparent meshes as they will no longer depth-test against themselves.
Yes, I was referring to that when I talked about "transparency artifacts".

Could also other objects nearer to the camera be hidden by a distant LDraw transparent mesh with `depthWrite=false`, depending on scene objects order?

**Comment by gkjohnson** [comment]
No matter how you cut it you're going to have some kind of artifact with transparent geometry unless you use a more complex approach -- my preference is to keep depthWrite to false when rendering transparent objects because I find the artifacts in that case to be less distracting.

**Comment by yomboprime** [comment]
> No matter how you cut it you're going to have some kind of artifact with transparent geometry unless you use a more complex approach -- my preference is to keep depthWrite to false when rendering transparent objects because I find the artifacts in that case to be less distracting.

I will try the following: leave `depthWrite` as before, but –as @WestLangley suggested– use `renderOrder` (lines and meshes are different objects so custom render order inside a Group should be possible. First render the lines and then the windscreens)

**Comment by WestLangley** [comment]
Should we remove the "temporary live links" in closed posts -- or not worry about it?

**Comment by yomboprime** [comment]
> Should we remove the "temporary live links" in closed posts -- or not worry about it?

As the branch gets deleted usually when the PR gets merged, I would say that the merge is the end of life of the links.

**Comment by mrdoob** [comment]
Calling them "temporary links" doesn't solve the issue?
Ideally this should be something we shouldn't have to worry about.

**Comment by WestLangley** [comment]
I find broken links annoying, so I prefer to delete the outdated temporary live links I create. But if others don't care, that's fine.

**Comment by mrdoob** [comment]
That sounds good to me 👌

**Pull request merged**